### PR TITLE
Fix Extended Data Query with no Access to Columns

### PR DESCRIPTION
**Pull request by mwizeman**
Updated to make sure that you have access to at least one column for a query to match extended data. Previously, if you had access to the element but none of the extended data columns you could still match on the row using a query with no filters. This resulted in the row being counted in `totalHits` but not loaded by Accumulo, which caused errors in the paging iterables.

### CODE REVIEW

File: elasticsearch5/src/main/java/org/vertexium/elasticsearch5/Elasticsearch5SearchIndex.java [hunk 1]
```diff
@@ -137,6 +140,26 @@ public Elasticsearch5SearchIndex(Graph graph, GraphConfiguration config) {
         this.logRequestSizeLimit = this.config.getLogRequestSizeLimit();
         this.exceptionHandler = this.config.getExceptionHandler(graph);
         this.flushObjectQueue = new FlushObjectQueue(this);
+
+        storePainlessScript("addFieldsToExtendedDataScript", getClass().getResourceAsStream("add-fields-to-extended-data.painless"), true);
```
Comment (joeferner) [inline elasticsearch5/src/main/java/org/vertexium/elasticsearch5/Elasticsearch5SearchIndex.java]: Move `getClass().getResourceAsStream` into `storePainlessScript`.

File: elasticsearch5/src/main/java/org/vertexium/elasticsearch5/Elasticsearch5SearchIndex.java [hunk 2]
```diff
@@ -137,6 +140,26 @@ public Elasticsearch5SearchIndex(Graph graph, GraphConfiguration config) {
...
+        storePainlessScript("updateFieldsOnDocumentScript", getClass().getResourceAsStream("update-fields-on-document.painless"), false);
+    }
+
+    private void storePainlessScript(String scriptId, InputStream scriptSource, boolean includeHelpers) {
+        try {
+            String source = IOUtils.toString(scriptSource);
+            if (includeHelpers) {
+                source = IOUtils.toString(getClass().getResourceAsStream("helper-functions.painless")) + " " + source;
+            }
+            source = source.replaceAll("\\r?\\n", " ");
```
Comment (joeferner) [inline elasticsearch5/src/main/java/org/vertexium/elasticsearch5/Elasticsearch5SearchIndex.java]: replace all `"` with `\"`

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,6 @@
+# v4.4.2
+* Fixed: If you had access an element but none of the extended data columns you could still match on the extended data row using a query with no filters. This would result in the row being counted in `totalHits`, but it could not be retrieved from Accumulo.
```
Comment (joeferner) [inline CHANGELOG.md]: Does this require a reindex? If so lets bump it to at least 4.5

File: elasticsearch5/src/main/java/org/vertexium/elasticsearch5/Elasticsearch5SearchIndex.java (hunk 1 above)
Comment (mwizeman) [reply elasticsearch5/src/main/java/org/vertexium/elasticsearch5/Elasticsearch5SearchIndex.java]: Fixed with: 575b8a8b960d6b77f8828e2c85bf1347503651c6

File: elasticsearch5/src/main/java/org/vertexium/elasticsearch5/Elasticsearch5SearchIndex.java (hunk 2 above)
Comment (mwizeman) [reply elasticsearch5/src/main/java/org/vertexium/elasticsearch5/Elasticsearch5SearchIndex.java]: Fixed with: 575b8a8b960d6b77f8828e2c85bf1347503651c6

File: CHANGELOG.md (hunk 3 above)
Comment (mwizeman) [reply CHANGELOG.md]: It does. Fixed with: 575b8a8b960d6b77f8828e2c85bf1347503651c6

**Pull request merged**